src/main/scala/com/amazon/deequ/analyzers/QuantileNonSample.scala

```diff
@@ -0,0 +1,307 @@
+   */
+  def update(item: T): Unit = {
+    compactors(0).buffer = compactors(0).buffer :+ item
+    compactorActualSize = compactorActualSize + 1
+    if (compactorActualSize > compactorTotalSize) {
+      condense()
+    }
+  }
+
+  def condense(): Unit = {
+    breakable {
+      for (height <- compactors.indices) {
+        if (compactors(height).buffer.length >= capacity(height)) {
+          if (height + 1 >= curNumOfCompactors) expand()
+          val output: Array[T] = compactors(height).compact
+          output.foreach(element => compactors(height + 1).buffer = compactors(height + 1).buffer :+ element)
+          compactorActualSize = getCompactorItemsCount
+          break
+        }
+      }
+    }
+  }
+
+  private def compareGreater[T](o1: T, o2: T)(implicit ord: Ordering[T]): Boolean = ord.gt(o1, o2)
+  private def compareSmaller[T](o1: T, o2: T)(implicit ord: Ordering[T]): Boolean = ord.lt(o1,o2)
+
+  /**
+   * Get the map which contains the rank of all items which is currently in the sketch.
+   * @return the sorted Listmap (by key)which contains the rank of all items which is currently in the sketch (RankMap)
+   */
+  def getRankMap(): ListMap[T,Long] = {
+    val sortedOutput = ListMap(output.toSeq.sortBy(_._1):_*)
+    var states = scala.collection.mutable.Map[T,Long]()
+    var runningRank = 0L
+    sortedOutput.foreach{ tuple =>
+      runningRank = runningRank + tuple._2
+      states(tuple._1) = runningRank
+    }
+    ListMap(states.toSeq.sortBy(_._1):_*)
+  }
+
+  /**
+   * Get CDF function of sketch items.
+   * @return CDF function
+   */
+  def getCDF(): Array[(T,Double)] = {
+    val rankMap = getRankMap()
+    val tmp = rankMap.keySet.toArray
+    val totalWeight = rankMap.last._2
+    var ret = ArrayBuffer[(T,Double)]()
+    tmp.foreach{ item =>
+      ret = ret :+ (item, rankMap(item).toDouble / totalWeight.toDouble)
+    }
+    ret.toArray
+  }
+
+  /**
+   * Get the rank of query item (inclusive rank) without RankMap.
+   * @param item item to query
+   * @return the estimated rank of the query item in sketch
+   */
+  def getRank(item:T): Long = {
+    var r = 0L
+    output.foreach(tuple => {
+      if (!compareGreater(tuple._1,item)) {
+        r = r + tuple._2
+      }
+    })
+    r
+  }
+
+  /**
+   * Get the rank of query item (exclusive rank) without RankMap.
+   * @param item item to query
+   * @return the estimated rank of the query item in sketch
+   */
+  def getRankExclusive(item:T): Long = {
+    var r = 0L
+    output.foreach(tuple => {
+      if (compareSmaller(tuple._1,item)) {
+        r = r + tuple._2
+      }
+    })
+    r
+  }
+
+  /**
+   * Get the rank of query item with RankMap.
+   * @param item item to query
+   * @param rankMap the estimated rank of the query item in sketch
+   * @return
+   */
+  def getRank(item:T, rankMap:ListMap[T,Long]): Long = {
+    var curRank = 0L
+    breakable {
+      for (target <- rankMap) {
+        if (compareGreater(target._1, item))
```

Comment:
please don't use `_1` and `_2`, destructure tuples and use descriptive variable names

Reply:
Updated.